server/src/test/java/org/elasticsearch/action/bulk/TransportBulkActionTests.java

```diff
@@ -321,31 +322,44 @@ public void testOnlySystem() {
         assertFalse(TransportBulkAction.isOnlySystem(buildBulkRequest(mixed), indicesLookup, systemIndices));
     }
 
-    public void testRejectCoordination() throws Exception {
+    private void blockWriteThreadPool(CountDownLatch blockingLatch) {
+        try {
+            final var executor = threadPool.executor(ThreadPool.Names.WRITE);
+            while (true) {
```

Comment:
This is a little suspicious. Should it eventually stop? Or should it spawn up to number of threads in the executor?

Reply:
It keeps going until the `executor.execute()` call throws an `EsRejectedExecutionException`. Let me add a comment.